app/uk/gov/hmrc/agentsubscriptionfrontend/connectors/AgentSubscriptionConnector.scala

```diff
@@ -148,6 +149,23 @@ class AgentSubscriptionConnector @Inject()(
         }
     }
 
+  def officerListContainsNameToMatch(crn: CompanyRegistrationNumber, name: String)(
+    implicit hc: HeaderCarrier): Future[Boolean] =
+    monitor(s"ConsumedAPI-Agent-Subscription-getCompanyOfficers-GET") {
+      val encodedCrn = UriEncoding.encodePathSegment(crn.value, "UTF-8")
+      val nameLowerCase = name.toLowerCase
+      val url =
+        s"${appConfig.agentSubscriptionBaseUrl}/agent-subscription/companies-house-api-proxy/company/$encodedCrn/officers/$nameLowerCase"
+      http
+        .GET[HttpResponse](url)
+        .map(_.status == 200)
+    }.recover {
+      case _: NotFoundException => {
+        Logger.warn(s"companies house returned Not Found for crn $crn and name: $name")
+        false
```

Comment:
i am just thinking BE can simply return true or false based on match or notMatch .. ?